src/app/packages/imagery/communicator-service/communicator.service.spec.ts

```diff
@@ -19,4 +22,156 @@ describe('CommunicatorEntity', () => {
+			Id: "1",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		const componentManager2: ImageryComponentManager = <any>{
+			Id: "2",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		imageryCommunicatorService.createCommunicator(componentManager1);
+		imageryCommunicatorService.createCommunicator(componentManager2);
+
+		const communicatorsObject = imageryCommunicatorService.communicators;
+		expect((<any>(communicatorsObject[componentManager1.Id]))._manager).toEqual(componentManager1);
+	});
+
+	it('communicatorsAsArray should return array of communicators', () => {
+
+		const componentManager1: ImageryComponentManager = <any>{
+			Id: "1",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		const componentManager2: ImageryComponentManager = <any>{
+			Id: "2",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		imageryCommunicatorService.createCommunicator(componentManager1);
+		imageryCommunicatorService.createCommunicator(componentManager2);
+
+		const communicatorsArray = imageryCommunicatorService.communicatorsAsArray();
+		expect(communicatorsArray.length).toEqual(2);
+	});
+
+	it('replaceCommunicatorId should replace the communicator Id', () => {
+
+		const componentManager1: ImageryComponentManager = <any>{
+			Id: "1",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		const componentManager2: ImageryComponentManager = <any>{
+			Id: "2",
+			centerChanged: new EventEmitter<GeoJSON.Point>(),
+			positionChanged: new EventEmitter<{id: string, position: MapPosition}>(),
+			pointerMove: new EventEmitter<any>(),
+			singleClick: new EventEmitter<any>()
+		};
+
+		imageryCommunicatorService.createCommunicator(componentManager1);
+		imageryCommunicatorService.createCommunicator(componentManager2);
+
+		imageryCommunicatorService.replaceCommunicatorId(componentManager2.Id, "3");
+
+		expect(imageryCommunicatorService.provide("2")).toEqual(null);
+		expect((<any>(imageryCommunicatorService.provide("3")))._manager).toEqual(componentManager2);
+	});
+
+	it('remove should remove the communicator from service', () => {
+
+		const componentManager1: ImageryComponentManager = <any>{
```

Comment:
componentManager1, componentManager2 can be initialize on 'describe' instead of every 'it'